src/cards/summon_generic.ts

```diff
@@ -27,181 +27,195 @@ import { DARK_PRIEST_ID } from '../entity/units/darkPriest';
+  [SUMMONER_ID]: {
+    exclude: false,
+    properties: { manaCost: 150 }
+  },
+  [GHOST_ARCHER_ID]: {
+    exclude: false,
+    properties: { manaCost: 180 }
+  },
+  [MANA_VAMPIRE_ID]: {
+    exclude: false,
+    properties: { manaCost: 180 }
+  },
+  [DARK_SUMMONER_ID]: {
+    exclude: false,
+    properties: { manaCost: 210 }
+  },
+  [DARK_PRIEST_ID]: {
+    exclude: false,
+    properties: { manaCost: 210 }
+  },
+  [bossmasonUnitId]: {
+    exclude: false,
+    properties: { manaCost: 400 }
+  },
+  'Spellmason': {
+    exclude: false,
+    properties: {}
+  },
 }
 export default function makeSpellForUnitId(unitId: string, asMiniboss: boolean, difficulty?: number): Spell | undefined {
-    const override = overrides[unitId];
-    const sourceUnit = allUnits[unitId];
-    if (!sourceUnit) {
-        console.error('Could not find source unit for ', unitId);
-        return undefined;
-    }
-    if (override && override.exclude) {
-        return undefined;
-    }
-    const unitAppearsAtLevelIndex = sourceUnit.spawnParams?.unavailableUntilLevelIndex || 1;
-    let rarity = CardRarity.COMMON;
-    if (unitAppearsAtLevelIndex < 4) {
-        rarity = CardRarity.COMMON
-    } else if (unitAppearsAtLevelIndex < 6) {
-        rarity = CardRarity.SPECIAL;
-    } else if (unitAppearsAtLevelIndex < 8) {
-        rarity = CardRarity.UNCOMMON;
-    } else if (unitAppearsAtLevelIndex < 10) {
-        rarity = CardRarity.RARE;
-    } else {
-        rarity = CardRarity.FORBIDDEN;
-    }
+  const override = overrides[unitId];
+  const sourceUnit = allUnits[unitId];
+  if (!sourceUnit) {
+    console.error('Could not find source unit for ', unitId);
+    return undefined;
+  }
+  if (override && override.exclude) {
+    return undefined;
+  }
+  const unitAppearsAtLevelIndex = sourceUnit.spawnParams?.unavailableUntilLevelIndex || 1;
+  let rarity = CardRarity.COMMON;
+  if (unitAppearsAtLevelIndex < 4) {
+    rarity = CardRarity.COMMON
+  } else if (unitAppearsAtLevelIndex < 6) {
+    rarity = CardRarity.SPECIAL;
+  } else if (unitAppearsAtLevelIndex < 8) {
+    rarity = CardRarity.UNCOMMON;
+  } else if (unitAppearsAtLevelIndex < 10) {
+    rarity = CardRarity.RARE;
+  } else {
+    rarity = CardRarity.FORBIDDEN;
+  }
 
-    const expenseScaling = 2;
-    const manaCost = override?.properties.manaCost ? (override.properties.manaCost * (asMiniboss ? 1.5 : 1)) : Math.max(1, Math.round(2 * Math.log2((sourceUnit.spawnParams?.budgetCost || 1)))) * 30 * (asMiniboss ? 2 : 1);
-    const id = unitId + (asMiniboss ? ' Miniboss' : '');
-    const unitSource = allUnits[unitId];
-    let healthMax = unitSource?.unitProps.healthMax || config.UNIT_BASE_HEALTH;
-    let manaMax = unitSource?.unitProps.manaMax || 0;
+  const expenseScaling = 2;
+  const manaCost = override?.properties.manaCost ? (override.properties.manaCost * (asMiniboss ? 1.5 : 1)) : Math.max(1, Math.round(2 * Math.log2((sourceUnit.spawnParams?.budgetCost || 1)))) * 30 * (asMiniboss ? 1.5 : 1);
```

Comment:
I think there's an override for every single summon now.  So this line shouldnt do anything other than use the override.  Why was this changed?

Reply:
Green glop was missing an override.

I also thought we'd want the miniboss multiplier to be the same for both overwritten and budget based summons